src/backend/storage/file/buffile.c

```diff
@@ -1122,7 +1152,7 @@ BufFileEndCompression(BufFile *file)
 		 file->uncompressed_bytes, file->maxoffset);
 
 	/* Done writing. Initialize for reading */
-	file->zstd_context->dctx = ZSTD_createDStream();
+	file->zstd_context->dctx = ZSTD_createDStream_advanced(customMem);
 	if (!file->zstd_context->dctx)
 		elog(ERROR, "out of memory");
 	ret = ZSTD_initDStream(file->zstd_context->dctx);
```

Comment:
Hmm... Where `ZSTD_endStream(file->zstd_context->dctx` ?

Reply:
`ZSTD_endStream` is for compression only and there's no equivalent function for decompression. `ZSTD_freeDCtx` is called in `BufFileClose` by `zstd_free_context`